### PR TITLE
libevm mega review

### PR DESCRIPTION
**Pull request by JcBernack**
- [x] [EVM-530](https://horizenlabs.atlassian.net/browse/EVM-530)
- [x] [EVM-529](https://horizenlabs.atlassian.net/browse/EVM-529)

### CODE REVIEW

**Review by akrckn — APPROVED** [review]

File: sdk/src/main/java/com/horizen/account/state/EvmMessageProcessor.java [hunk 1]
```diff
@@ -32,11 +32,15 @@ public byte[] process(Message msg, BaseAccountStateView view, GasPool gas, Block
             throws ExecutionFailedException {
         // prepare context
         var context = new EvmContext();
+        // TODO: set correct ChainID
```
Comment (i-Alex) [inline sdk/src/main/java/com/horizen/account/state/EvmMessageProcessor.java:35]: I this TODO relevant?
I think we setup it properly in all the places now.

**Review by i-Alex — COMMENTED** [review]
LGTM!

File: sdk/src/main/java/com/horizen/account/state/EvmMessageProcessor.java (hunk 1 above)
Comment (JcBernack) [reply sdk/src/main/java/com/horizen/account/state/EvmMessageProcessor.java:35]: Oops, yes. Outdated comment.

Comment (JcBernack) [reply sdk/src/main/java/com/horizen/account/state/EvmMessageProcessor.java]: done

**Review by i-Alex — APPROVED** [review]

**Pull request merged**